sharktank/tests/models/llama/sharded_llama_test.py

```diff
@@ -346,91 +286,105 @@ def _(model, *args, **kwargs) -> torch.Tensor:
-                *prefill_iree_cache_state_shards
-            )
-            for actual_cache_state_shard, expected_cache_state_shard in zip(
-                prefill_iree_cache_state_shards,
-                sharded_prefill_args["cache_state"][0].shards,
-            ):
-                # TODO: debug inaccuracy.
-                torch.testing.assert_close(
-                    actual_cache_state_shard, unbox_tensor(expected_cache_state_shard)
-                )
+        # Run prefill step.
+        prefill_iree_args = prepare_iree_module_function_args(
+            args=deepcopy(sharded_prefill_args).values(), devices=iree_devices
+        )
+        for i, arg in enumerate(prefill_iree_args):
+            np.save(f"{path_prefix}prefill_arg{i}.npy", arg.to_host())
+        prefill_iree_result = run_iree_module_function(
+            args=prefill_iree_args,
+            function_name="prefill",
+            module=iree_module,
+            vm_context=vm_context,
+            driver=iree_driver,
+            trace_path_prefix=path_prefix if dump_enabled else None,
+        )
+        prefill_iree_result = iree_to_torch(*prefill_iree_result)
+        assert len(prefill_iree_result) == 1
+        expected_prefill_result = call_torch_module_function(
+            module=sharded_model,
+            function_name="prefill",
+            kwargs=sharded_prefill_args,
+            trace_path_prefix=f"{path_prefix}expected_" if dump_enabled else None,
+        )
+        prefill_iree_cache_state_shards = prefill_iree_args[
+            -self.config.tensor_parallelism_size - 1 :
+        ]
+        prefill_iree_cache_state_shards = iree_to_torch(
+            *prefill_iree_cache_state_shards
+        )
 
-            # Check IREE's decode step is close to torch.
-            decode_iree_args = prepare_iree_module_function_args(
-                args=deepcopy(sharded_decode_args).values(), devices=iree_devices
-            )
-            decode_iree_result = run_iree_module_function(
-                args=decode_iree_args,
-                function_name="decode",
-                module=iree_module,
-                vm_context=vm_context,
+        # Run decode step.
+        decode_iree_args = prepare_iree_module_function_args(
+            args=deepcopy(sharded_decode_args).values(), devices=iree_devices
+        )
+        decode_iree_result = run_iree_module_function(
+            args=decode_iree_args,
+            function_name="decode",
+            module=iree_module,
+            vm_context=vm_context,
+            driver=iree_driver,
+            trace_path_prefix=path_prefix if dump_enabled else None,
+        )
+        decode_iree_result = iree_to_torch(*decode_iree_result)
+        expected_decode_result = call_torch_module_function(
+            module=sharded_model,
+            function_name="decode",
+            kwargs=sharded_decode_args,
+            trace_path_prefix=f"{path_prefix}expected_" if dump_enabled else None,
```

Comment:
You should use `os.path.join` for combining

Reply:
This prefix is interpreted as a string, not as a directory. This allows to prefix not only with a directory, but a part of the filename as well.